docs/managing-e2e-tests.md

```diff
@@ -0,0 +1,166 @@
+
+## Managing kubeadm end-to-end tests
+
+### Overview
+
+Kubernetes has a complicated end-to-end (e2e) testing infrastructure, which allows detailed testing of clusters and their assets. Most settings and tools for that can be found in the [test-infra](https://github.com/kubernetes/test-infra) GitHub repository.
+
+Kubernetes uses applications such as the web-based [testgrid](https://k8s-testgrid.appspot.com/) for monitoring the status of e2e tests. test-infra also host the configuration on individual test jobs and Docker images that contain tools to invoke the jobs.
+
+### Details on kubeadm test jobs
+
+The following folder contains all the SIG Cluster Lifecycle (the SIG that maintains kubeadm) originated test jobs:
+[sig-cluster-lifecycle](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-cluster-lifecycle)
+
+Please note that this document will only cover details on the `kubeadm*.yaml` files.
+
+For example, let's have a look at this file:
+[kubeadm.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm.yaml)
+
+It contains a list of jobs such as:
+```
+- name: ci-kubernetes-e2e-kubeadm-gce-master
+  interval: 2h
+  ...
+```
+
+In this case, `ci-kubernetes-e2e-kubeadm-gce-master` is a test job that runs every 2 hours and it also has a set of parameters defined for it. This document will only cover some of them.
+
+Important to note here is that test-grid jobs have two important assets, a `deployment` and a docker `image` that contains all the tools needed to run tests.
+
+For this example job the `deployment` is called [kubernetes-anywhere](https://github.com/kubernetes/kubernetes-anywhere) (with integration in [this file](https://github.com/kubernetes/test-infra/blob/master/kubetest/anywhere.go)) and the `image` is called `gcr.io/k8s-testimages/e2e-kubeadm:...`. As a very rough summary, the way this works is when a job is invoked all the job parameters are passed to a CLI tool called `kubetest` which then passes them to the deployment tool.
```

Comment:
s very rough/very high level

Might be rephrase "which then passes them to the deployment tool." In "which then instantiate the job container and then the deployment tool inside it"